audio-worklet/noisegate-test.html

```diff
@@ -0,0 +1,506 @@
+          // Similarly, the first release weight is 1 / transitionSteps.
+          expectedReleaseWeights[j] =  (j + 1) / transitionSteps; 
+        }
+        // Allow for a small error (set experimentally) since the exact change 
+        // in weight is computed differently here and in noisegate.computeGain.
+        const errorThreshold = 1e-6;
+        should(attackWeights, 'Attacking weights')
+            .beCloseToArray(expectedAttackWeights, {absoluteThreshold: 1e-6});
+        should(releaseWeights, 'Releasing weights')
+            .beCloseToArray(expectedReleaseWeights, {absoluteThreshold: 1e-6});
+        should(closedWeights, 'Weight array before threshold')
+            .beConstantValueOf(0);
+        should(openWeights, 'Weight array after the threshold')
+            .beConstantValueOf(1);
+        task.done();
+      }); 
+
+    audit.run(); 
+    
+    // End of test cases. Testing utility functions below!
+
+    function _getAveragedAmplitudes(signal, numberOfAmplitudes) {
+      // GoingUp is true if current value is greater than last value.
+      let amplitudes = new Array();
+      let goingUp = signal[1] > signal[0] ? true : false;
+      // Peak is where transitions goingUp changes state and amplitude is recorded.
+      let peak = signal[0];
+      for (let i = 2; i < signal.length; i++) {
+        if (signal[i] <  signal[i-1]) {
+          if (goingUp) {
+            amplitudes.push(Math.abs(signal[i] - peak));
+            peak = signal[i];
+            goingUp = false;
+          }
+        } else {
+          if (!goingUp) {
+            amplitudes.push(Math.abs(signal[i] - peak));
+            peak = signal[i];
+            goingUp = true;
+          }
+        }
+      }
+
+      if (amplitudes.length < numberOfAmplitudes)
+        throw 'Cannot request ' + numberOfAmplitudes +
+            ' since there were only ' + amplitudes.length +
+            ' distinct amplitudes in the signal.';
+
+      // Return an array of specified number of amplitudes, 
+      // where each amplitude represents an average of adjacent amplitudes. 
+      numberOfAmplitudes++; 
+      const amplitudesToAverage = Math.floor(amplitudes.length / numberOfAmplitudes);
+      averagedAmplitudes = new Array();
+
+      // Start at 1 since first amplitude is shortened.
+      for (let i = 1; i < amplitudes.length - amplitudesToAverage; i+= amplitudesToAverage) {
+        let sum = 0;
+        for (let j = 0; j < amplitudesToAverage; j++ ) {
+          sum += amplitudes[j+i];
+        }
+        averagedAmplitudes.push(sum / amplitudesToAverage);
+      }
+
+      return averagedAmplitudes;
+    }
+
+    function _toIndex(time, sampleRate) {
+      if (sampleRate == null || time == null) throw ('Illegal parameters');
+      return Math.floor(time * sampleRate);
+    }
+
+    function _toTime(index, sampleRate) {
+      if (sampleRate == null || index == null) throw ('Illegal parameters');
+      return index * (1 / sampleRate);
```

Comment:
I put them in there for my own benefit. I always forgot to add the sample rate :)